### PR TITLE
Allow for docker capabilities to be added from cyborg

### PR DESCRIPTION
**Pull request by Mark-Powers**
Any linux capabilities included in the OCI config from cyborg will be included in the docker host config. The specific capability sets are ignored, if a capability appears in any of these, it is added to the container.

### CODE REVIEW

File: zun/container/docker/driver.py [hunk 1]
```diff
@@ -343,6 +343,10 @@ def create(self, context, container, image, requested_networks,
                     # TODO: actually look up cgroups for access info
                     host_config['devices'].append(
                         f'{dev["path"]}:{dev["path"]}:rw')
+                host_config["cap_add"] = []
+                for cap_list in oci_config.get("capabilities", {}).values():
```
Comment (diurnalist) [inline zun/container/docker/driver.py:347]: Could be a one-liner using `itertools.chain`, which takes a list of lists and returns one list consisting of a "chain" of all lists you gave it, e.g.: `chain(['A', 'B', 'C'], ['D, 'E', 'F']) => ['A', 'B', 'C', 'D', 'E', 'F']`:

```python
# Using 'set' to trim duplicates
host_config["cap_add"] = set(itertools.chain(oci_config.get("capabilities", {}).values())
```

More readable (no more one-liner ;))

```python
capabilities = oci_config.get("capabilities", {})
all_caps = itertools.chain(capabilities.values())
cap_add = list(set(all_caps))
```

File: zun/container/oci.py [hunk 2]
```diff
@@ -16,10 +16,14 @@
 
 
 def merge_oci_runtime_config(parent, *cfgs):
+    cap_sets = ["bounding", "permitted", "inheritable", "effective", "ambient"]
```
Comment (diurnalist) [inline zun/container/oci.py:19]: Probably worth pulling out as a constant, and citing https://man7.org/linux/man-pages/man7/capabilities.7.html

File: zun/container/oci.py [hunk 3]
```diff
@@ -30,6 +34,14 @@ def merge_oci_runtime_config(parent, *cfgs):
         cfg_devices = cfg.get('linux', {}).get('devices')
         if cfg_devices is not None:
             devices.extend(cfg_devices)
+        cfg_capabilities = cfg.get('capabilities', {})
+        if cfg_capabilities is not None:
```
Comment (diurnalist) [inline zun/container/oci.py:38]: How can this be `None`? Maybe better to make the above line `cfg.get('capabilities') or {}` instead, so it handles if it's explicitly set to None.

**Review by diurnalist — APPROVED** [review]
🙇🏻

**Review by msherman64 — APPROVED** [review]
Looks alright to me. Thanks!

**Pull request merged**